tools/generator/src/Generator+SetTargetConfigurations.swift

```diff
@@ -285,56 +352,88 @@ $(CONFIGURATION_BUILD_DIR)
     }
 
     private static func handleTestHost(
-        for target: Target,
+        for target: ConsolidatedTarget,
         disambiguatedTargets: DisambiguatedTargets,
-        pbxTargets: [TargetID: PBXTarget],
+        pbxTargets: [ConsolidatedTarget.Key: PBXTarget],
         attributes: inout [String: Any],
-        buildSettings: inout [String: BuildSetting]
+        buildSettings: inout [BuildSettingConditional: [String: BuildSetting]]
     ) throws {
-        if let testHostID = target.testHost {
-            guard
-                let testHost = disambiguatedTargets.targets[testHostID]?.target
-            else {
-                throw PreconditionError(message: """
-Test host target with id "\(testHostID)" not found in `disambiguatedTargets`
+        let targets = target.targets.values
+
+        guard let aTestHostID = targets.first?.testHost else {
```

Comment:
Should this find _any_ test host, not just the one set on the first target?

E.g. `targets.lazy.compactMap(\.testHost).first`